sql/src/main/java/io/crate/metadata/SystemTable.java

```diff
@@ -171,9 +172,14 @@ public String toString() {
 
         private final ArrayList<Column<T, ?>> columns = new ArrayList<>();
         private List<ColumnIdent> primaryKeys = List.of();
-        private Function<DiscoveryNodes, Routing> getRouting;
+        private BiFunction<DiscoveryNodes, RoutingProvider, Routing> getRouting;
 
         public RelationBuilder<T> withRouting(Function<DiscoveryNodes, Routing> getRouting) {
```

Comment:
I'm wondering if we should remove this overload and force all users of SystemTable to use the other variant.

By keeping this, we have another indirection, due to the `(node, routingProvider) -> getRouting.apply(node)`.

Reply:
I am also not a big fan of overloading in general, i would say let's remove the `Function<DiscoveryNodes, Routing>` variant.